lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -979,8 +983,34 @@ def restart_all_services(self):
             )
         elif self._init_system == InitMode.DOCKER:
             self.exec_command("cd /home/vagrant/magma/lte/gateway/docker && docker-compose restart")
-        print("Waiting for all services to restart. Sleeping for 60 seconds..")
-        self.wait_for_restart_to_finish(60)
+
+        self._wait_for_pipelined_to_initialize()
+
+    def _wait_for_pipelined_to_initialize(self):
+        print("Waiting for pipelined to be started ...")
+        INITIAL_WAIT_TIME_SECONDS = 10
+
+        print(f"  waiting initially {INITIAL_WAIT_TIME_SECONDS} seconds ...")
```

Comment:
We were earlier waiting for 60 seconds for all services to come up because some of them were taking more than 15 seconds in general and sometimes seen to take even around 40 seconds. Here we are setting 10 seconds initial wait time. Is this sufficient for all services to come up? 

Because as per this change as soon as pipelined is initialized it will move ahead without waiting any more even if other services dependent on pipelined (like mme service) are initialized or not.

Considering the above corner case, I think the better approach would have been 
- to remove the initial wait time here (as that is not very helpful, since pipelined would be rarely initialized within that initial wait time of 10 seconds)
- to add an additional 10 seconds wait time after we see the pipelined is initialized to give some buffer for other services which are dependent on pipelined to be ready.

Reply:
We implemented your approach, thanks!